jsonargparse_tests/test_dataclass_like.py

```diff
@@ -587,6 +718,37 @@ def test_pydantic_types(self, valid_value, invalid_value, cast, type_str, monkey
             parser.parse_args([f"--model.param={invalid_value}"])
         ctx.match("model.param")
 
+    @pytest.mark.skipif(not pydantic_supports_field_init, reason="Field.init is required")
+    def test_dataclass_field_init_false(self, parser):
+        # This tests the following error:
+        #
+        # TypeError: Parser key "data.p2":
+        # Expected a <class 'str'>. Got value: annotation=str
+        # required=False default='-' init=False
```

Comment:
What is this? How does this test a TypeError? If an error is tested, then I would expect an `pytest.raises` instead of some confusing comment.